packages/@webex/plugin-cc/src/AgentLogin/AgentLogin.ts

```diff
@@ -0,0 +1,38 @@
+import {HTTP_METHODS, WebexSDK} from '../types';
+
+export default class AgentLogin {
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(webex: WebexSDK, wccAPIURL: string) {
+    this.webex = webex;
+    this.wccAPIURL = wccAPIURL;
+  }
+
+  public async loginAgentWithSelectedTeam(
+    teamId: string,
+    agentDeviceType: string,
+    deviceId: string
+  ) {
+    try {
+      const body = {
+        dialNumber: deviceId,
+        teamId,
+        isExtension: agentDeviceType === 'EXTENSION',
+        roles: ['agent'],
+        deviceType: agentDeviceType,
+        deviceId,
+      };
+      const loginResponse = await this.webex.request({
+        method: HTTP_METHODS.POST,
+        uri: `${this.wccAPIURL}v1/agents/login`,
+        body,
+      });
+      this.webex.logger.log('LOGIN API INVOKED');
+
+      return loginResponse;
```

Comment:
This has to be Promise.resolve(response) and please name it as response. Also add the error handling for failed responses. And promise resolve should be dependent on event reception